src/HotChocolate/Core/src/Types/Resolvers/IMiddlewareContext.cs

```diff
@@ -1,12 +1,77 @@
 using System;
+using System.Collections;
 using System.Collections.Generic;
+using System.Linq;
 using System.Threading.Tasks;
 using HotChocolate.Types;
 
 #nullable enable
 
 namespace HotChocolate.Resolvers;
 
+public struct TypedValueT<TLogicalElementType>
+{
+    public TypedValueT(object value, bool isCollection)
+    {
+        Value = value;
+        IsCollection = isCollection;
+    }
+
+    public bool IsCollection { get; }
+    public object Value { get; }
+    public readonly Type LogicalElementType => typeof(TLogicalElementType);
```

Comment:
> a `readonly` member? any particular reason for this?

Readonly on getters means it doesn't mutate the struct. Well, since there are no mutable fields, might as well just make the whole struct readonly.

Reply:
agree, i was wondering why only LogicalElementType is marked as readonly. Readonly in structs is funny anyway as the compiler does some funny thing there.